blog/tests.py

```diff
@@ -1,3 +1,176 @@
+
+        def test_loggedout(self):
+            "If user is logged out they wont be able to edit posts"
+            self.client.logout()
+            author = User.objects.create (username = "hello_there101")
+            published_post = Post.objects.create(title = "Post1", text = "Post 1 data", author = author, published_date = now())
+            response = self.client.get(f"/post/{published_post.pk}/edit/")
+            self.assertEqual(response.status_code, 404) #making sure its sucessful
+            unpublished_post = Post.objects.create(title = "Post2", text = "Post 2 data", author = author)
+            response = self.client.get(f"/post/{unpublished_post.pk}/edit/")
+            self.assertEqual(response.status_code, 404)
+
+        def test_post_published(self):
+            """If post published user should be able to edit it sucessfully"""
+            author = User.objects.create (username = "hello_there101") #created the author
+            published_post = Post.objects.create(title = "Post1", text = "Post 1 data", author = author, published_date = now())
+            response = self.client.post(f"/post/{published_post.pk}/edit/", {"title":"new title","text":"new text"})
+            self.assertRedirects(response,f"/post/{published_post.pk}/") #making sure its sucessful
+            published_post.refresh_from_db() #makes sure to get the most recent data from the data base
+            self.assertEqual(published_post.title, "new title")
+            self.assertEqual(published_post.text, "new text")
+
+        def test_post_unpublished(self):
+            """If post in unpublished user should not be able to edit it sucessfully"""
+            author = User.objects.create(username = "user1")
+            unpublished_post = Post.objects.create(title = "Post02", text = "Great", author = author)
+            response = self.client.post(f"/post/{unpublished_post.pk}/edit/", {"title": "new title", "text": "new text"})
+            self.assertRedirects(response, f"/post/{unpublished_post.pk}/")
+            unpublished_post.refresh_from_db()
+            self.assertEqual(unpublished_post.title, "new title")
+            self.assertEqual(unpublished_post.text, "new text")
+
+        def test_post_loggedout(self):
+            """If user is logged out they wont be able to edit posts"""
+            self.client.logout()
+            author = User.objects.create (username = "hello_there101")
+            published_post = Post.objects.create(title = "Post1", text = "Post 1 data", author = author, published_date = now())
+            response = self.client.post(f"/post/{published_post.pk}/edit/")
+            self.assertEqual(response.status_code, 404) #making sure its sucessful
+            unpublished_post = Post.objects.create(title = "Post2", text = "Post 2 data", author = author)
+            response = self.client.post(f"/post/{unpublished_post.pk}/edit/")
+            self.assertEqual(response.status_code, 404)
```

Comment:
After this test, could you also add a test for trying to edit a post that doesn't exist (for example, with a URL of `"/post/1000000/edit/"` or `"/post/!!!!!/edit/"`)?